### PR TITLE
feat(1091): add endpoint to list template versions and associated popularity metrics

### PR DESCRIPTION
**Pull request by InderH**
## Context

Need to return associated popularity metrics with template versions so the metrics can be displayed on the templates page.

## Objective

This PR adds an endpoint that returns template versions along with their associated popularity metrics.

## References

https://github.com/screwdriver-cd/screwdriver/issues/1091

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: test/plugins/templates.test.js [hunk 1]
```diff
@@ -387,6 +389,56 @@ describe('template plugin test', () => {
...
+
+        beforeEach(() => {
+            options = {
+                method: 'GET',
+                url: '/templates/screwdriver%2Fbuild/metrics'
+            };
+        });
+
+        it('returns 200 and all template versions and metrics for a template name', () => {
+            templateFactoryMock.listWithMetrics.resolves(testtemplateversionsmetrics);
```
Comment (klu909) [inline test/plugins/templates.test.js]: it's better to add this line `assert.expect(x);` with `x` being number of asserts in your test

Comment (InderH) [reply test/plugins/templates.test.js]: assert.expect() does not exist in this repo

File: test/plugins/templates.test.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ const testtemplate = require('./data/template.json');
 const testtemplates = require('./data/templates.json');
 const testtemplatetags = require('./data/templateTags.json');
 const testtemplateversions = require('./data/templateVersions.json');
+const testtemplateversionsmetrics = require('./data/templateVersionsMetrics.json');
```
Comment (tkyi) [inline test/plugins/templates.test.js]: camelcase would be better
```suggestion
const testTemplateVersionsMetrics = require('./data/templateVersionsMetrics.json');
```

Comment (InderH) [reply test/plugins/templates.test.js]: updated it, thanks

**Pull request merged**